spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/TestSparkDistributionAndOrderingUtil.java

```diff
@@ -299,14 +300,70 @@ public void testRangeWritePartitionedSortedTable() {
     checkWriteDistributionAndOrdering(table, expectedDistribution, expectedOrdering);
   }
 
+  // =============================================================
+  // Distribution and ordering for copy-on-write DELETE operations
+  // =============================================================
+  //
+  // UNPARTITIONED UNORDERED
+  // -------------------------------------------------------------------------
+  // delete mode is NOT SET -> CLUSTER BY _file + LOCALLY ORDER BY _file, _pos
+  // delete mode is NONE -> unspecified distribution + empty ordering
+  // delete mode is HASH -> CLUSTER BY _file + LOCALLY ORDER BY _file, _pos
+  // delete mode is RANGE -> CLUSTER BY _file + LOCALLY ORDER BY _file, _pos
+  //
+  // UNPARTITIONED ORDERED BY id, data
+  // -------------------------------------------------------------------------
+  // delete mode is NOT SET -> CLUSTER BY _file + LOCALLY ORDER BY _file, _pos
```

Comment:
I am debating a bit what should be the correct ordering in this case given that our table has a defined sort order. On one hand, sorting by `_file` and `_pos` may lead to unsorted files on S3 if the object storage paths are enabled. On the other hand, sorting by the defined sort columns may rebalance files (could be a good thing?).

Reply:
Does `ORDERED BY` above mean that we have the table distribution mode set to `range`? If so, I'd probably do a global order as though the delete mode were set to `range`. I think that makes sense since the order is requested.

Reply:
No, it did not mean that but I may need to take it into account too. Let me think.

Reply:
Well, I am not sure I agree on this one. I think regular writes and deletes are fundamentally different and it is our best interest to avoid a range-based shuffle unless the user explicitly asked for it during deletes. I think our default should be to keep the existing order with the minimal cost and a hash shuffle would be way cheaper. What do you think?

This would be also a change in the behavior compared to what we released before.

Reply:
That said, I think we should order by `id, data` in this case to have properly sorted files. If there are two files with id (0, 10) and (2, 11) and they end up in one writing task, we probably want to have a properly sorted file.

Reply:
The comment in your last reply sounds good to me.